sdk/identity/Azure.Identity/src/TokenCacheUpdatedArgs.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// Data regarding an update of a <see cref="TokenCache"/>.
+    /// </summary>
+    public class TokenCacheUpdatedArgs
+    {
+        internal TokenCacheUpdatedArgs(TokenCache cache)
+        {
+            Cache = cache;
+        }
+
+        /// <summary>
+        /// The <see cref="TokenCache"/> instance which was updated.
+        /// </summary>
+        public TokenCache Cache { get; }
```

Comment:
Won't this always be the instance to which you've subscribed to?

Reply:
Yes but you could have a single handler subscribing to the updated events of multiple caches, (this is actually one of the main use cases)  so we pass back the instance which is firing the event. We could have used the EventHandler "object sender" pattern, but decided to add it as a property to the event args instead to make the signature simpler as well as type safe.

Reply:
You can always capture the instance you are subscribing to though.

cache.Updated += async () DoSomethingWith(cache);

Reply:
But maybe the Args type is worth to have just from the perspective of future proofing.

Reply:
Yes it's quite likely that the data we pass to this event will grow. There's other information that we could pass to this event to help optimize serialization, but for the first draft we elected to keep it simple.